app/src/main/java/com/ikarabulut/Board.java

```diff
@@ -22,6 +23,7 @@ public Board(int numOfRows, int numOfColumns) {
         grid = setGrid();
     }
 
+    // PUBLIC METHODS
     public List<ArrayList> setGrid() {
```

Comment:
I still think we're generating a board here, not setting a grid, but let's see what Jeff & Claudia think too.
```suggestion
    public List<ArrayList> generate() {
```

Similarly, I'd rename `setRandomGrid()` to `generateRandomBoard()` and `getGrid() to `getBoard()`

Reply:
Prefer to keep domain terms distinct. Using different terms for the same thing leads to code that is confusing to read.  Is it a Board or a Grid?

Also, I think John is right in that generate is a better description for what `setGrid` is doing.

When I read Java code, I expect the following conventions:

* A method name with the prefix of `get` (`getFoo`) returns a field on the object named 'foo'.
* A method name with a prefix of `set` (`setFoo(value)`) sets the field `foo` to the value.

Together these conventions are known as "getters" and "setters".  

Technically the `setGrid` function is setting a value for a field named `grid`.  But it feels like it's doing a lot more than a little validation on an input value.  Also, it's not taking a value to set anything to.  This is why the word `generate` is better for describing what this method is doing.

Reply:
Good points all around. Updated to board, and generate. Shown in my next push to this branch